plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotIntegrationSmokeTest.java

```diff
@@ -0,0 +1,634 @@
+    private static final int DEFAULT_PINOT_LIMIT_FOR_BROKER_QUERIES = 10;
+    private static final int ALL_TYPES_TABLE_ROW_COUNT = 1000;
+    private static final int ALL_TYPES_TABLE_NULL_ROW_COUNT = 17;
+    private static final int ALL_TYPES_TABLE_ARRAY_NULL_ROW_COUNT = 13;
+    private static final String MIXED_CASE_TABLE = "mixed_case";
+    private static final List<List<Object>> MIXED_CASE_TABLE_RESULTS;
+    private static final String JSON_TABLE = "my_table";
+
+    static {
+        ImmutableList.Builder<List<Object>> mixedCaseResultSetBuilder = ImmutableList.builder();
+        for (long i = 0; i <= MAX_ROWS_PER_SPLIT; i++) {
+            mixedCaseResultSetBuilder.add(ImmutableList.builder()
+                    .add("string_" + i)
+                    .add(i)
+                    .add(Instant.now().truncatedTo(ChronoUnit.DAYS).plusMillis(i).toEpochMilli())
+                    .build());
+        }
+        MIXED_CASE_TABLE_RESULTS = mixedCaseResultSetBuilder.build();
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Random random = new Random(22);
+        TestingKafka kafka = closeAfterClass(TestingKafka.createWithSchemaRegistry());
+        kafka.start();
+        TestingPinotCluster pinot = closeAfterClass(new TestingPinotCluster(kafka.getNetwork()));
+        pinot.start();
+
+        // Create and populate the all_types topic and table
+        kafka.createTopic(ALL_TYPES_TABLE);
+
+        int step = 3;
+        int offset = 1;
+        checkState(ALL_TYPES_TABLE_ROW_COUNT > ALL_TYPES_TABLE_NULL_ROW_COUNT, "COUNT must be greater than NULL_COUNT");
+        checkState(ALL_TYPES_TABLE_NULL_ROW_COUNT > ALL_TYPES_TABLE_ARRAY_NULL_ROW_COUNT, "NULL_COUNT must be greater than ARRAY_NULL_COUNT");
+        ImmutableList.Builder<ProducerRecord<String, GenericRecord>> builder = ImmutableList.builder();
+        for (long i = 0; i < ALL_TYPES_TABLE_ROW_COUNT - ALL_TYPES_TABLE_NULL_ROW_COUNT; i++) {
+            builder.add(new ProducerRecord<>(ALL_TYPES_TABLE, "key" + i * step,
+                    createTestRecord(
+                            random.ints(3, 0, 10001).limit(3).mapToObj(val -> "string_" + val).collect(toList()),
+                            Arrays.asList(random.nextBoolean(), random.nextBoolean(), random.nextBoolean()),
+                            random.ints().limit(3).boxed().collect(toList()),
+                            random.ints(3, -10000, 10001).asDoubleStream().map(d -> d / 10000).boxed().map(Double::floatValue).collect(toList()),
+                            random.ints(3, -10000, 10001).asDoubleStream().map(d -> d / 10000).boxed().collect(toList()),
+                            random.longs(3, ((long) Integer.MIN_VALUE) * 2, ((long) Integer.MAX_VALUE) * 2).boxed().collect(toList()),
+                            Instant.now().truncatedTo(ChronoUnit.DAYS).plusMillis(offset + i * step).toEpochMilli())));
```

Comment:
use a fixed point in time to stay clear from any DST issues that we're not (consciously) testing for